src/league_cmd.cpp

```diff
@@ -0,0 +1,170 @@
+ */
+std::tuple<CommandCost, LeagueTableID> CmdCreateLeagueTable(DoCommandFlag flags, const std::string &title, const std::string &header, const std::string &footer)
+{
+	if (!LeagueTable::CanAllocateItem()) return { CMD_ERROR, INVALID_LEAGUE_TABLE };
+	if (title.empty()) return { CMD_ERROR, INVALID_LEAGUE_TABLE };
+
+	if (flags & DC_EXEC) {
+		LeagueTable *lt = new LeagueTable();
+		lt->title = title;
+		lt->header = header;
+		lt->footer = footer;
+		return { CommandCost(), lt->index };
+	}
+
+	return { CommandCost(), INVALID_LEAGUE_TABLE };
+}
+
+
+/**
+ * Create a new element in a league table.
+ * @param flags type of operation
+ * @param table Id of the league table this element belongs to
+ * @param rating Value that elements are ordered by
+ * @param company Company to show the color blob for or INVALID_COMPANY
+ * @param text Text of the element
+ * @param score String representation of the score associated with the element
+ * @param link_type Type of the referenced object
+ * @param link_target Id of the referenced object
+ * @return the cost of this operation or an error
+ */
+std::tuple<CommandCost, LeagueTableElementID> CmdCreateLeagueTableElement(DoCommandFlag flags, LeagueTableID table, int64 rating, CompanyID company, const std::string &text, const std::string &score, LinkType link_type, LinkTargetID link_target)
+{
+	if (!LeagueTableElement::CanAllocateItem()) return { CMD_ERROR, INVALID_LEAGUE_TABLE_ELEMENT };
+	Link link{link_type, link_target};
+	if (!IsValidLink(link)) return { CMD_ERROR, INVALID_LEAGUE_TABLE_ELEMENT };
+	if (company != INVALID_COMPANY && !Company::IsValidID(company)) return { CMD_ERROR, INVALID_LEAGUE_TABLE_ELEMENT };
+
+	if (flags & DC_EXEC) {
+		LeagueTableElement *lte = new LeagueTableElement();
+		lte->table = table;
+		lte->rating = rating;
+		lte->company = company;
+		lte->text = text;
+		lte->score = score;
+		lte->link = link;
+		InvalidateWindowData(WC_COMPANY_LEAGUE, table);
+		return { CommandCost(), lte->index };
+	}
+	return { CommandCost(), INVALID_LEAGUE_TABLE_ELEMENT };
+}
+
+/**
+ * Update the attributes of a league table element.
+ * @param flags type of operation
+ * @param element Id of the element to update
+ * @param company Company to show the color blob for or INVALID_COMPANY
+ * @param text Text of the element
+ * @param link_type Type of the referenced object
+ * @param link_target Id of the referenced object
+ * @return the cost of this operation or an error
+ */
+CommandCost CmdUpdateLeagueTableElementData(DoCommandFlag flags, LeagueTableElementID element, CompanyID company, const std::string &text, LinkType link_type, LinkTargetID link_target)
+{
+	if (_current_company != OWNER_DEITY) return CMD_ERROR;
+	auto lte = LeagueTableElement::GetIfValid(element);
+	if (lte == nullptr) return CMD_ERROR;
+
+	Link link{link_type, link_target};
+	if (!IsValidLink(link)) return CMD_ERROR;
+	if (company != INVALID_COMPANY && !Company::IsValidID(company)) return CMD_ERROR;
```

Comment:
This seems grouped in an odd place with `link` checks. Maybe it belongs in the section above, near the check that enforces `_current_company == OWNER_DEITY`?